packages/js/client/src/__tests__/core/error-structure.spec.ts

```diff
@@ -186,53 +168,56 @@ describe("Error structure", () => {
 
       expect(result.error?.innerError instanceof WrapError).toBeTruthy();
       const prev = result.error?.innerError as WrapError;
       expect(prev.name).toEqual("WrapError");
       expect(prev.code).toEqual(WrapErrorCode.WRAPPER_INVOKE_ABORTED);
       expect(prev.reason.startsWith("SubInvocation exception encountered")).toBeTruthy();
-      expect(prev.uri).toEqual("wrap://ens/bad-math.eth");
-      expect(prev.method).toEqual("subInvokeWillThrow");
-      expect(prev.args).toEqual("{\n  \"0\": 130,\n  \"1\": 161,\n  \"2\": 97,\n  \"3\": 1,\n  \"4\": 161,\n  \"5\": 98,\n  \"6\": 1\n}");
+      expect(prev.uri).toEqual("wrap://ens/imported-invoke.eth");
+      expect(prev.method).toEqual("invokeThrowError");
+      expect(prev.args).toEqual("{\n  \"0\": 129,\n  \"1\": 161,\n  \"2\": 97,\n  \"3\": 163,\n  \"4\": 72,\n  \"5\": 101,\n  \"6\": 121\n}");
       expect(prev.source).toEqual({ file: "~lib/@polywrap/wasm-as/containers/Result.ts", row: 171, col: 13 });
 
       expect(prev.innerError instanceof WrapError).toBeTruthy();
       const prevOfPrev = prev.innerError as WrapError;
       expect(prevOfPrev.name).toEqual("WrapError");
       expect(prevOfPrev.code).toEqual(WrapErrorCode.WRAPPER_INVOKE_ABORTED);
-      expect(prevOfPrev.reason).toEqual("__wrap_abort: I threw an error!");
-      expect(prevOfPrev.uri.endsWith("wrap://ens/bad-util.eth")).toBeTruthy();
-      expect(prevOfPrev.method).toEqual("iThrow");
-      expect(prevOfPrev.args).toEqual("{\n  \"0\": 129,\n  \"1\": 161,\n  \"2\": 97,\n  \"3\": 0\n}");
-      expect(prevOfPrev.source).toEqual({ file: "src/index.ts", row: 5, col: 5 });
+      expect(prevOfPrev.reason).toEqual("__wrap_abort: Hey");
+      expect(prevOfPrev.uri.endsWith("wrap://ens/imported-subinvoke.eth")).toBeTruthy();
+      expect(prevOfPrev.method).toEqual("subinvokeThrowError");
+      expect(prev.args).toEqual("{\n  \"0\": 129,\n  \"1\": 161,\n  \"2\": 97,\n  \"3\": 163,\n  \"4\": 72,\n  \"5\": 101,\n  \"6\": 121\n}");
+      expect(prevOfPrev.source).toEqual({ file: "src/index.ts", row: 8, col: 5 });
     });
 
-    test("Invoke a wrapper of incompatible version", async () => {
-      const result = await client.invoke<string>({
-        uri: incompatibleWrapperUri.uri,
-        method: "simpleMethod",
-        args: {
-          arg: "test",
-        },
-      });
-
-      expect(result.ok).toBeFalsy();
-      if (result.ok) throw Error("should never happen");
-
-      expect(result.error?.name).toEqual("WrapError");
-      expect(result.error?.code).toEqual(WrapErrorCode.URI_RESOLVER_ERROR);
-      expect(result.error?.uri.endsWith("packages/test-cases/cases/wrappers/wasm-as/simple-deprecated")).toBeTruthy();
-      expect(result.error?.resolutionStack).toBeDefined();
-      expect(`${result.error?.cause}`).toContain(`Unrecognized WrapManifest schema version "0.0.1"`);
-    });
+    // test("Invoke a wrapper of incompatible version", async () => {
```

Comment:
i will try that! or maybe just modifying an existing manifest with another version just to test that it indeed throws

Reply:
sounds like a nice solution

Reply:
i just did an update but it is not working, when I modify the manifest and try to create a wasm wrapper (`WasmWrapper.from(...)`) it fails, because the version is not right (it does a validation on the creation of a wasm wrapper, which I think make sense) - Also tried with a plugin but the invocation passes, which probably means that we have a problem there. Will debug more on Monday but if you want to take a look feel free to :smile:

Reply:
We should assume the wrapper was created with a previous version of the toolchain, and that's why it exists. I think we just need a wrap manifest with an old version, and to try to invoke it with the filesystem resolver.

Reply:
you can use the wrapper that already exists in `origin-dev` that i was using for the test before

Reply:
Regarding the plugin, we might want to try to import an old plugin made with prealpha polywrap and see what happens. There might be a problem before an invocation can even be made. Maybe we should create an issue to explore this more?

Reply:
> We should assume the wrapper was created with a previous version of the toolchain, and that's why it exists. I think we just need a wrap manifest with an old version, and to try to invoke it with a filesystem uri

it worked as expected, you're a genius :1st_place_medal:, please check the test and let me know if this make sense :smile: 

> Regarding the plugin, we might want to try to import an old plugin made with prealpha polywrap and see what happens. There might be a problem before an invocation can even be made. Maybe we should create an issue to explore this more?

i went ahead and create a test cases (which is currently skipped), so I'd say it make sense to create the issue (feel free to check my test to make sure I am thinking it in the right way)